File: test/normalize_employee_name_test.dart
import 'package:challenges/src/normalize_employee_name.dart';
import 'package:test/test.dart';

void main() {
  test("it should return empty and whitespaces string for empty input", () {
    expect(normalizeEmployeeName(" "), "");
    expect(normalizeEmployeeName("    "), "");
    expect(normalizeEmployeeName("          "), "");
  });
  test("it should return capitalize every first word for input", () {
    expect(normalizeEmployeeName("jaswanth kumar"), "Jaswanth Kumar");
    
  });
  test("it should handle for whitespace ", () {
    expect(normalizeEmployeeName("jaswanth    kumar"), "Jaswanth Kumar");
  });
  test("it should handle for case  ", () {
   expect(normalizeEmployeeName("jaswAnth kuMar"), "Jaswanth Kumar");
  });
}

File: javachallenges/src/test/java/challenges/NormalizeEmployeeNameTest.java
package challenges;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class NormalizeEmployeeNameTest {
    @Test
    void testEmpty(){
        assertEquals(NormalizeEmployeeName.NormalizeEmployeeName(""),  "invalid input");
        assertEquals(NormalizeEmployeeName.NormalizeEmployeeName(" "),  "invalid input");
        assertEquals(NormalizeEmployeeName.NormalizeEmployeeName("  "),  "invalid input");
    }
    @Test
    void test(){
        assertEquals(NormalizeEmployeeName.NormalizeEmployeeName("jaswanth kumar"), "Jaswanth Kumar");
    }
    @Test
    void testNameswithWhitespaces(){
        assertEquals(NormalizeEmployeeName.NormalizeEmployeeName("jasWanth     kumar"), "Jaswanth Kumar");
    }
}
